Update destructuring 2.md

SRGOM opened the pull request:
I'm very new to Rust and the changes I proposed may either be incorrect or not fit in the flow of the document. Please feel free to suggest modifications that I can make or to reject altogether. Thanks.

nrc commented on destructuring 2.md:
```diff
@@ -45,14 +45,10 @@ semantics:
+You can also choose to have copy semantics for structs defined by you. 
+Use `#[derive(Copy)]` to implement the `Copy` trait and the struct would 
+have Copy semantics. You may also need to derive the `Clone` trait needed
+by the `Copy` trait.
```
This is an improvement but it is not quite complete. So, I would say that to have copy semantics you have to implement `Copy` and `derive` is one way to do that. It is also worth noting that there restrictions on whether a type can have the `Copy` trait implemented - in particular, the destructor thing (see https://doc.rust-lang.org/std/marker/trait.Copy.html#when-can-my-type-be-copy for details).

SRGOM commented:
Thanks you for taking the time to comment (and of course the very wonderfully written series). I (think I) have addressed your comment and put back some destructor related content from my previous deletion. Please let me know if any more changes are required.

nrc commented on destructuring 2.md:
```diff
@@ -45,14 +45,14 @@ semantics:
-fields are examined and if any of those do then the whole object has move
-semantics. And so on down the object structure. If no destructors are found
-anywhere in an object, then it has copy semantics.
+You can also choose to have copy semantics for user-defined structs
```
s/structs/types

nrc commented on destructuring 2.md:
```diff
@@ -45,14 +45,14 @@ semantics:
-anywhere in an object, then it has copy semantics.
+You can also choose to have copy semantics for user-defined structs
+by implementing the `Copy` trait. One straightforward way to do that is 
+to add `#[derive(Copy)]` before the definition of the struct. Not all
```
and here

nrc replied:
and below in a few places

nrc commented on destructuring 2.md:
```diff
@@ -45,14 +45,14 @@ semantics:
+those  with a destructor. Destructors probably need a post of their own,
+but for now, an object in Rust has a destructor if it implements the `Drop`
+trait. Just like C++, the destructor is executed just before an object is 
+destroyed. If an object has a destructor then it has move semantics.
```
I would drop this last sentence

SRGOM commented:
I've addressed your comments, please let me know if any other changes are required.

nrc commented:
Thanks for the changes, looks good!
